integration/test/ca_bundle/ca_bundle_test.go

```diff
@@ -0,0 +1,114 @@
+const targetString = "x509: certificate signed by unknown authority"
+//Let the agent run for 1 minutes. This will give agent enough time to call server
+const agentRuntime = 1
+
+
+type input struct {
+	findTarget bool
+	dataInput string
+}
+
+//Must run this test with parallel 1 since this will fail if more than one test is running at the same time
+func TestBundle(t *testing.T) {
+
+	parameters := []input{
+		{dataInput: "resources/integration/ssl/with/combine/bundle", findTarget: false},
+		{dataInput: "resources/integration/ssl/with/original/bundle", findTarget: true},
+		{dataInput: "resources/integration/ssl/without/bundle", findTarget: true},
+	}
+
+	for _, parameter := range parameters {
+		//before test run
+		log.Printf("resource file location %s find target %t", parameter.dataInput, parameter.findTarget)
+		clearLogFile()
+		t.Run(fmt.Sprintf("resource file location %s find target %t", parameter.dataInput, parameter.findTarget), func(t *testing.T) {
+			copyFile(parameter.dataInput + configJSON, configOutputPath)
+			copyFile(parameter.dataInput + commonConfigTOML, commonConfigOutputPath)
+			startTheAgent();
+			time.Sleep(agentRuntime * time.Minute);
+			log.Printf("Agent has been running for : %d", agentRuntime);
+			stopTheAgent();
+			containsTarget := readTheOutputLog()
+			if (parameter.findTarget && !containsTarget) || (!parameter.findTarget && containsTarget) {
+				t.Errorf("Find target is %t contains target is %t", parameter.findTarget, containsTarget)
+			}
+		})
+	}
+}
+
+func clearLogFile() {
+	cmd := exec.Command("bash", "-c", "sudo rm " + outputLog + " && sudo touch " + outputLog)
+	err := cmd.Run()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Printf("Cleared log file : %s", outputLog);
+}
+
+func copyFile(pathIn string, pathOut string) {
+	log.Printf("Copy File %s to %s", pathIn, pathOut)
+	pathInAbs, err := filepath.Abs(pathIn)
+	log.Printf("File %s abs path %s", pathIn, pathInAbs)
+	if err != nil {
+		log.Fatal(err)
+	}
+	cmd := exec.Command("bash", "-c", "sudo cp " + pathInAbs + " " + pathOut)
+	err = cmd.Run()
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.Printf("File : %s copied to : %s", pathIn, pathOut);
+}
+
+func startTheAgent() {
+	cmd := exec.Command("bash", "-c", "sudo /opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a " +
+		"fetch-config -m ec2 -s -c file:" +
+		configOutputPath)
+	err := cmd.Run()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Printf("Agent has started");
+}
+
+func stopTheAgent() {
+	cmd := exec.Command("bash", "-c", "sudo /opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a stop")
+	err := cmd.Run()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Printf("Agent is stopped");
+}
+
+func readTheOutputLog() bool {
+	logFile, err := ioutil.ReadFile(outputLog)
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.Printf("Log file %s", string(logFile))
+	log.Printf("Finished reading log file")
```

Comment:
Do we need this log? wouldn't the log above it already prove that we finished reading the log file? Separately, I don't think this needs to be a `Printf()` because there's nothing to format in there.